internal/controllers/addon_controller.go

```diff
@@ -91,8 +101,10 @@ func (r *AddonReconciler) Reconcile(
 	// Phase 4.
 	// Ensure CatalogSource for this Addon
 	{
+		// TODO dont pass logger, internal to Reconciler struct
```

Comment:
The log instance here is bound to a particular reconcile request and is created at the very start of this method. We might want to create our own `RequestContext` to pass around here, but that's another discussion.